[dev ex] Provide `docs/source/tutorial` for faster local documentation build (#2277)
optuna/optuna: CONTRIBUTING.md

To build the documentation, you need to run:

```bash

> pip install -e ".[document]" -f https://download.pytorch.org/whl/torch_stable.html

Reviewer: C.f. https://github.com/optuna/optuna/pull/2305

HTML files are generated under `build/html` directory. Open `index.html` with the browser and see
if it is rendered as expected.

Optuna's tutorial is built with [Sphinx-Gallery](https://sphinx-gallery.github.io/stable/index.html) and
some other requirements like [LightGBM](https://github.com/microsoft/LightGBM) and [PyTorch](https://pytorch.org) meaning that
all .py files in `tutorial` directory are run during the documentation build if there's no build cache.
Whether you edit any tutorial or not doesn't matter.

To avoid having to run the tutorials, you may download executed tutorial artifacts nanmed "tutorial" from our CI (see the capture below) and put them in `docs/build` before
extract the files in the zip to `docs/source/tutorial` directory.

> ![image](https://user-images.githubusercontent.com/16191443/107472296-0b211400-6bb2-11eb-9203-e2c42ce499ad.png)

Reviewer: 👍

pytest tests/${TARGET_TEST_FILE_NAME}
```

> ## Documentation

Reviewer: There is a `### Documentation` section above, explaining `make html`, etc. How about keeping this content closer to `make html` since they're related? I'm also thinking that we could move the entire above section into this one. We don't need two.

Optuna's tutorial is built with [Sphinx-Gallery](https://sphinx-gallery.github.io/stable/index.html) and
some other requirements like [LightGBM](https://github.com/microsoft/LightGBM) meaning that
all .py files in `tutorial` directory are run during the documentation build if there's no build cache.
Whether you edit any tutorial or not doesn't matter.

> To avoid the run of tutorial, we recommend that you download build artifacts from our CI and put them to `docs/build`.

Reviewer: "Recommend" is maybe a bit strong, at least without further guidance to where and what exactly to download.
```suggestion
To avoid having to run the tutorials, you may download build artifacts from our CI and put them in `docs/build`.
```
